spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/MessageListenerContainer.java

```diff
@@ -57,4 +58,39 @@ default boolean isConsumerBatchEnabled() {
 		return false;
 	}
 
+	/**
+	 * Set the queue names.
+	 * @param queues the queue names.
+	 * @since 2.4
+	 */
+	default void setQueueNames(String... queues) {
+		throw new UnsupportedOperationException();
+	}
+
+	/**
+	 * Set auto startup.
+	 * @param autoStart true to auto start.
+	 * @since 2.4
+	 */
+	default void setAutoStartup(boolean autoStart) {
+	}
+
+	/**
+	 * Get the message listener.
+	 * @return The message listener object.
+	 * @since 2.4
+	 */
+	@Nullable
+	default Object getMessageListener() {
+		return null;
+	}
+
+	/**
+	 * Set the listener id.
+	 * @param id the id.
+	 * @since 2.4
+	 */
+	default void setListenerId(String id) {
```

Comment:
Why do we need to have all these methods in `default` state?
Shouldn't target impl provide them anyway?

Reply:
I didn't want to make breaking changes, but I guess you are right.